### PR TITLE
MM-15758 Update dependencies including Fastlane

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR updates all dependencies but `react-intl` and `react-native-notifications` that have their own tickets.

doc-viewer was replaced by file-viewer as it seems doc-viewer is no longer maintained

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-15758

### CODE REVIEW

**Comment by enahum** [comment]
@chuttam @migbot please this one is sort of urgent?

File: patches/react-native-webview+9.4.0.patch [hunk 1]
```diff
@@ -0,0 +1,188 @@
...
++
+     @Override
+     public void onPageFinished(WebView webView, String url) {
+       super.onPageFinished(webView, url);
+@@ -810,6 +823,49 @@ public class RNCWebViewManager extends SimpleViewManager<WebView> {
+         new TopLoadingErrorEvent(webView.getId(), eventData));
+     }
+ 
++    @Override
++    public void onReceivedHttpAuthRequest(WebView view,
```
Comment (chuttam) [inline patches/react-native-webview+9.4.0.patch:68]: Is this Alert new functionality for Mattermost?

Comment (chuttam) [reply patches/react-native-webview+9.4.0.patch:68]: If was already part of Mattermost, can you point me to where it came from?

File: app/init/global_event_handler.js [hunk 2]
```diff
@@ -236,28 +237,30 @@ class GlobalEventHandler {
...
-                        text: translations[t('mobile.server_upgrade.button')],
-                        onPress: this.serverUpgradeNeeded,
-                    }],
-                    {cancelable: false},
-                );
-            } else if (state.entities.users && state.entities.users.currentUserId) {
-                dispatch(setServerVersion(serverVersion));
-                const data = await dispatch(loadConfigAndLicense());
-                this.configureAnalytics(data.config);
+        if (Store?.redux?.dispatch) {
```
[the remark this answers is not in the document]
Comment (enahum) [reply app/init/global_event_handler.js:240]: this was there before, so we were for sure hitting this issue, so I just wanted to be on the safe side, no harm in that

File: patches/react-native-webview+9.4.0.patch (hunk 1 above)
Comment (enahum) [reply patches/react-native-webview+9.4.0.patch:68]: We were using a fork before https://github.com/mattermost/react-native-webview/ this is to help with IWA authentication

**Comment by enahum** [comment]
@josephbaylon do any of your findings happens on Master? just to confirm

**Comment by josephbaylon** [comment]
> @josephbaylon do any of your findings happens on Master? just to confirm

@enahum I also tested on both master branch and v1.31 build 293 and they are not happening there

**Comment by enahum** [comment]
Thanks for confirming

**Comment by josephbaylon** [comment]
> Initial issues found.
> 
> 1. On Android, when user edits a post and tries to save it, the app crashes
> 2. On Android, when user tries to open Edit Profile, the app crashes
> 3. On Android, when user tries to open Edit Channel, the app crashes
> 4. On both Android and iOS, on first install, Flagged Posts are empty. However, on succeeding open/close app and re-installations, Flagged Posts screen shows the list.

5. On Android, when user taps on plus(+) for Direct Messages, Private Channels, Public Channels,  the app crashes

**Comment by enahum** [comment]
Ok @josephbaylon there was a bug in the react-native-navigations library so I've downgraded to the version in master for now

**Comment by enahum** [comment]
RNN issue can be found here https://github.com/wix/react-native-navigation/issues/6204

**Comment by josephbaylon** [comment]
@enahum I retested the issues above and they seem to fixed now. However, I encountered 3 intermittent issues, which I couldn't reproduce consistently but happens once in a while. I tested these scenarios on v1.31 build 293 and they don't happen there at all.

Happens on Android only.
1. Logout of server fast
- Login to community server
- Before Reception channel fully loads, tap on RHS menu and logout

Observed: Sometimes an error prompt appears once user is logged out 
image-902e4934-8ea9-4b43-89dd-93270361f2bf

2. Sometimes channel is stuck on skeleton frame (see image below)
image-400ab123-e355-47a7-8008-f132e5ba5c60

3. Sometimes switching servers doesn't work. This is the hardest to reproduce.
- Login to mysql.test server then logout
- Try to login to community server

Observed: user is still logged in to mysql.test server

**Comment by enahum** [comment]
@josephbaylon 
> I tested these scenarios on v1.31 build 293 and they don't happen there at all

The one displaying the alert box does happen. Reason: The session is revoked before the WS is able to connect. The reality is that this would be rarely hit as having the WS failing to authenticate while the session is still active is very rare, I'm going to simple remove the alert.

#2 I haven't been able to reproduce and there is this ticket https://mattermost.atlassian.net/browse/MM-24372 more so it has been reported by customers.. so if you can help with repro steps on this one.. that would be AMAZING!!!

**Comment by enahum** [comment]
For #3 it seems like a race condition that we cannot avoid unless there is a MAJOR refactor. If you  logout while there are requests that haven't completed, once they do, they will populate data even if it is from another server. The likelihood for this is very low, but I do agree this is something that can happen. We do not cancel any request that is in progress while logging out nor prevent that to be stored. If you think this is a use case that we should cover I would ask to file a ticket.

The other option we have is to reset the state once you tap on the `Connect` button -> I'll add this just in case.

**Pull request merged**